File: client/go/src/example/connect/ssl.go
package connect

import (
	"fmt"
	"context"
	"biostar/service/connect"
)

func (s *ConnectSvc) EnableSSL(deviceIDs []uint32) error {
	req := &connect.EnableSSLMultiRequest{
		DeviceIDs:      deviceIDs,
	}

	resp, err := s.client.EnableSSLMulti(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot enable SSL: %v %v", err, resp)
		return err
	}

	return nil
}


func (s *ConnectSvc) DisableSSL(deviceIDs []uint32) error {
	req := &connect.DisableSSLMultiRequest{
		DeviceIDs:      deviceIDs,
	}

	resp, err := s.client.DisableSSLMulti(context.Background(), req)

	if err != nil {
		fmt.Printf("Cannot disable SSL: %v %v", err, resp)
		return err
	}

	return nil
}

File: client/node/example/connect/ssl.js
const connectMessage = require('../../biostar/service/connect_pb');
const connect = require('./connect');

function enableSSL(deviceIDs) {
  var req = new connectMessage.EnableSSLMultiRequest();
  req.setDeviceidsList(deviceIDs);

  return new Promise((resolve, reject) => {
    connect.getClient().enableSSLMulti(req, (err, response) => {
      if(err) {
        console.error('Cannot enable SSL: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}


function disableSSL(deviceIDs) {
  var req = new connectMessage.DisableSSLMultiRequest();
  req.setDeviceidsList(deviceIDs);

  return new Promise((resolve, reject) => {
    connect.getClient().disableSSLMulti(req, (err, response) => {
      if(err) {
        console.error('Cannot disable SSL: ', err)
        reject(err);
        return;
      }

      resolve(response);
    });
  });
}


module.exports.enableSSL = enableSSL;
module.exports.disableSSL = disableSSL;
